### PR TITLE
Dropdown motion update dynamic duration

### PR DESCRIPTION
**Pull request by wonilsuh**
## Overview

Applies System 360 motion specs to the Dropdown component. This includes both the new curves and the duration scale.

Also adds three more variables to the globals.

### Added

```
// src/globals/scss/_vars
$transition--expansion: 150ms;
$transition--fade: 100ms;
$transition--delay-unit:200ms;
```

### Changed

Both the JS and SCSS were changed.

```
src/components/dropdown/dropdown.js
src/components/dropdown/dropdown.scss
```

## Testing / Reviewing

`$ npm run test`

### CODE REVIEW

**Comment by tw15egan** [comment]
This PR is out of date with changes from master. Can you rebase off of master and push up the changes?

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "lodash.debounce": "^4.0.8"
   },
   "devDependencies": {
+    "@ibm/motion": "^0.2.1",
```
Comment (asudoh) [inline package.json]: Should be in `dependencies`

File: src/components/dropdown/dropdown.js [hunk 2]
```diff
@@ -87,19 +88,41 @@ class Dropdown extends mixin(createComponent, initComponentBySearch, trackBlur)
...
       const actions = {
         add: isOfSelf && event.which === 40 && !isOpen,
         remove: (!isOfSelf || event.which === 27) && isOpen,
         toggle: isOfSelf && event.which !== 27 && event.which !== 40,
       };
+      const shouldOpen = actions.add === true || (actions.toggle === true && isOpen !== true);
+      const onTransitionEnd = () => {
+        listEl.removeEventListener('transitionend', onTransitionEnd);
+        listEl.style.height = '';
+        // listEl.style.transitionDuration = '';
```
Comment (asudoh) [inline src/components/dropdown/dropdown.js]: Please remove this commented-out line.

File: src/components/dropdown/dropdown.js [hunk 3]
```diff
@@ -87,19 +88,41 @@ class Dropdown extends mixin(createComponent, initComponentBySearch, trackBlur)
...
+        if (shouldOpen === true) listEl.classList.add('bx--dropdown-list--open');
+        else listEl.classList.remove('bx--dropdown-list--open');
+      };
+      const w = element.ownerDocument.defaultView;
       Object.keys(actions).forEach(action => {
         if (actions[action]) {
-          this.element.classList[action]('bx--dropdown--open');
-          this.element.focus();
+          element.classList[action]('bx--dropdown--open');
+          element.focus();
```
Comment (asudoh) [inline src/components/dropdown/dropdown.js]: Keyboard focus should be done after transition ends.

File: src/components/dropdown/dropdown.js [hunk 4]
```diff
@@ -179,7 +202,7 @@ class Dropdown extends mixin(createComponent, initComponentBySearch, trackBlur)
    * Closes the dropdown menu if this component loses focus.
    */
   handleBlur() {
-    this.element.classList.remove('bx--dropdown--open');
+    this._toggle({ type: 'blur' });
```
Comment (asudoh) [inline src/components/dropdown/dropdown.js]: Kind of nit-picky, but `handleBlur()` takes actual event in the first arg, and I’d rather use it for the sake of consistency. The event has `focusin` or `focus` type.

**Review by asudoh — APPROVED** [review]
@wonilsuh Great that you came up with a much simpler version! Let’s see if this is something the team can agree on - I think this simpler version has higher bet.

In addition to my comment to the code, the separated open/closes states (CSS classes) for `.bx—dropdown-list` has to be reflected to the test cases, given that most of them tests that all necessary CSS classes are updated upon user action, etc. Thanks!

**Review by asudoh — CHANGES_REQUESTED** [review]
Oops I made a wrong selection - Changes are needed to get it through, sorry!

**Review by asudoh — CHANGES_REQUESTED** [review]
Thanks @wonilsuh for making the updates! Seems that some of changes in https://github.com/carbon-design-system/carbon-components/pull/357/files and https://github.com/carbon-design-system/carbon-components/pull/362/files are still lost, great to have you take a look at them - Thanks!

**Comment by wonilsuh** [comment]
@asudoh Thanks for checking the files - can you pull again please? I do see the files on both local and on Git.

**Comment by wonilsuh** [comment]
@asudoh now i understand the problems, fixing them. Thank you again!

**Pull request closed without merging**